src/ATen/native/xpu/sycl/LogcumsumexpKernel.cpp

```diff
@@ -0,0 +1,177 @@
+#include <ATen/Dispatch.h>
+#include <ATen/OpMathType.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/core/TensorBase.h>
+
+#include <ATen/native/xpu/sycl/ScanKernels.h>
+#include <ATen/native/xpu/sycl/ScanUtils.h>
+
+#include <cmath>
+#include <limits>
+
+namespace at::native::xpu {
+
+static c10::MaybeOwned<Tensor> contiguous_out_arg(const Tensor& tensor) {
+  if (tensor.is_contiguous()) {
+    return c10::MaybeOwned<Tensor>::borrowed(tensor);
+  }
+  return c10::MaybeOwned<Tensor>::owned(
+      at::empty(tensor.sizes(), tensor.options()));
+}
```

Comment:
merge with LogcumsumexpKernel.cpp, CumsumKernel.cpp and rename as ScanKernels.cpp, aligned with CUDA

Reply:
Thanks for review. Done.